system_tests/system_tests_sync/test_external_accounts.py

```diff
@@ -136,3 +139,68 @@ def test_file_based_external_account(
                 },
             },
         )
+
+
+# This test makes sure that setting up an http server to provide credentials
+# works to allow access to Google resources.
+def test_url_based_external_account(dns_access, oidc_credentials, service_account_info):
+    class TestResponseHandler(BaseHTTPServer.BaseHTTPRequestHandler):
+        def do_GET(self):
+            if self.headers["my-header"] != "expected-value":
```

Comment:
I think we should also confirm the correct path is accessed: `/token`
I think you can parse it from `self.path`?

Reply:
done